src/crimson/osd/exceptions.h

```diff
@@ -11,6 +11,10 @@ struct crimson_error : public std::system_error {
     : system_error(std::make_error_code(ec)) {
   }
 
+  static crimson_error from_intret(const int ret) {
```

Comment:
how about
```c++
namespace ceph::osd {
  class error  {
    /// ...
    static error from_err(int e) {
      return error{-e};
    }
  };
}
```

or
```c++
namespace ceph::osd {
  class error  {
    /// ...
    friend error make_error(int ret) {
      return error{-ret};
    }
  };
}
```

Reply:
I think I like the `make_error` concept very much!

Reply:
Added the namespace.